app/uk/gov/hmrc/serviceconfigs/service/ConfigService.scala

```diff
@@ -104,14 +104,18 @@ class ConfigService @Inject()(
                                 case ("backend.conf", v)  if optBootstrapFile.contains("backend.conf")  => "bootstrapBackendConf"  -> ConfigParser.parseConfString(v)
                               }
                               .toMap
-      applicationConfWithoutIncludes = ConfigParser.parseConfString(applicationConfRaw, Map.empty, logMissing = false)
-      applicationConf = // collecting the config without includes helps `delta` identify what was explicitly included in application.conf
-                        // however, in the rare case that a substitution refers to something in bootstrapConf, we will need to consider it to resolve,
-                        // but it will lead to thinking that all bootstrapConf entries also exist explicitly in applicationConf
-                        if (Try(applicationConfWithoutIncludes.resolve()).isFailure)
-                          ConfigParser.parseConfString(applicationConfRaw, ConfigParser.toIncludeCandidates(referenceConfigs))
-                        else
-                          applicationConfWithoutIncludes
+      includedAppConfig  = optSlugInfo.fold(Map.empty[String, String])(_.includedAppConfig) // combine any local config split out from application.conf
+      applicationConf    = // collecting the config without dependency configs helps `delta` identify what was explicitly included in application.conf
+                           // however, in the rare case that a substitution refers to something in bootstrapConf, we will need to consider it to resolve,
+                           // but it will lead to thinking that all bootstrapConf entries also exist explicitly in applicationConf
+                           { val applicationConfWithoutDependencyConfig = ConfigParser.parseConfString(applicationConfRaw, includedAppConfig, logMissing = false)
```

Comment:
I think this would be simpler if defined in the outer for-comprehension

Reply:
thanks .... outered